contracts/libs/Witnet.sol

```diff
@@ -3,314 +3,370 @@
-  function asInt128Array(WitnetData.Result memory _result) external pure returns(int128[] memory) {
-    require(_result.success, "Tried to read `int128[]` value from errored WitnetData.Result");
-    return _result.value.decodeInt128Array();
-  }
-
-  /// @notice Decode a string value from a WitnetData.Result as a `string` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string` decoded from the WitnetData.Result.
-  function asString(WitnetData.Result memory _result) external pure returns(string memory) {
-    require(_result.success, "Tried to read `string` value from errored WitnetData.Result");
-    return _result.value.decodeString();
-  }
-
-  /// @notice Decode an array of string values from a WitnetData.Result as a `string[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string[]` decoded from the WitnetData.Result.
-  function asStringArray(WitnetData.Result memory _result) external pure returns(string[] memory) {
-    require(_result.success, "Tried to read `string[]` value from errored WitnetData.Result");
-    return _result.value.decodeStringArray();
-  }
-
-  /// @notice Decode a natural numeric value from a WitnetData.Result as a `uint64` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64` decoded from the WitnetData.Result.
-  function asUint64(WitnetData.Result memory _result) external pure returns(uint64) {
-    require(_result.success, "Tried to read `uint64` value from errored WitnetData.Result");
-    return _result.value.decodeUint64();
-  }
+    /// Data kept in EVM-storage for every Request posted to the Witnet Request Board.
+    struct Request {
+        IWitnetRequest addr;    // The contract containing the Data Request which execution has been requested.
+        address requestor;      // Address from which the request was posted.
+        bytes32 codehash;       // Codehash of the Data Request which execution has been requested.
+        uint256 gasprice;       // Minimum gas price the DR resolver should pay on the solving tx.
+        uint256 reward;         // escrow reward to by paid to the DR resolver.
+    }
 
-  /// @notice Decode an array of natural numeric values from a WitnetData.Result as a `uint64[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64[]` decoded from the WitnetData.Result.
-  function asUint64Array(WitnetData.Result memory _result) external pure returns(uint64[] memory) {
-    require(_result.success, "Tried to read `uint64[]` value from errored WitnetData.Result");
-    return _result.value.decodeUint64Array();
-  }
+    /// Data kept in EVM-storage containing Witnet-provided response metadata and result.
+    struct Response {
+        address reporter;       // Address from which the result was reported.
+        bytes32 proof;          // Witnet-provided validation proof of the reported result.
```

Comment:
is this a proof or is it the transaction hash of the request? With proof is the PoI that comes to my mind

Reply:
In general terms, it's meant to be a `bytes32` that enables the application to verify result correctness (and finality) of provided result within the context of the Witnet blockchain. In practice, in the centralized implementation: the hash used as "drTxHash" before.

Reply:
The thing is that when the time comes to use an actual merkle proof here, it won't fit into `bytes32`, so it's a bit strange that we are thinking long-term but only half-way.

I can't imagine either any version of the request board that uses this field for anything else that fits into `bytes32`. I believe we can assume for certain that any future version of the request board that has a different trust model will necessarily require a new version of the `Response` structure that is laid out differently. If something, I can imagine that we may add a new field containing an actual proof, and the old field would be left empty or would be repurposed, so having this named `proof` may turn out to be confusing.

At this point, I think we are totally free to be explicit about what this represents. Otherwise we are just "hiding" the true nature of this field, right?

Reply:
Certain as we are that an eventual merkle proof would not fit within `bytes32` field, let's better name it as what it is in practive: "witnetTxHash", or simply "txhash" ?